src/partitioner/histogram.py

```diff
@@ -0,0 +1,39 @@
+"""Utilities for generating and manipulating object count histograms"""
+
+import healpy as hp
+import numpy as np
+
+from partitioner.io_utils import read_dataframe
+
+
+def empty_histogram(highest_order):
+    """Use numpy to create an histogram array with the right shape, filled with zeros"""
+    return np.zeros(hp.order2npix(highest_order), dtype=np.ulonglong)
+
+
+def generate_partial_histogram(
+    file_path,
+    highest_order,
+    file_format,
+    ra_column,
+    dec_column,
+):
+    """Generate a histogram of counts for objects found in the indicated file_path"""
+    histo = empty_histogram(highest_order)
+    data = read_dataframe(file_path, file_format)
+
+    # Verify that the file has columns with desired names.
+    if not all([x in data.columns for x in [ra_column, dec_column]]):
```

Comment:
so from googling around this is actually the pythonic way to check that a list contains multiple values but at a glance I had trouble figuring out what this line did 😭 . Might I suggest either changing this to
```suggestion
    if ra_column not in data.columns or dec_column not in data.columns:
```
which is slightly more verbose but clearer, or creating a `required_columns = [ra_column, dec_column]` list above and changing this to
```suggestion
if not all([x in data.columns for x in required_columns]):
```
which is also more explicit and would have the benefit of allowing us to expand `required_columns` if we need to in the future.

Reply:
Thanks! I like the second option a little better, since we might be expanding the list.